code/parse/sexp.cpp

```diff
@@ -13320,6 +13321,37 @@ void sexp_player_use_ai(int flag)
 	}
 }
 
+void sexp_set_friendly_damage_caps(int n) {
+	bool is_nan;
+	bool is_nan_forever;
+	ai_profile_t& aip = *The_mission.ai_profile;
+
+	float beam_friendly_cap = i2fl(eval_num(n, is_nan, is_nan_forever));
+	for (auto &bfc : aip.beam_friendly_damage_cap) {
+		bfc = beam_friendly_cap;
+	}
```

Comment:
This will set damage caps for *all* difficulty levels at once.  Consider some other options:
1. Set the caps for only the current difficulty level
2. Add an argument to set the caps for only the specified difficulty level
3. Use multiple arguments and allow the FREDder to specify multiple caps for the different difficulty levels
4. Add other SEXP operators to allow setting caps independently
5. Some combination of the above

I think if I were doing this, I'd make three SEXP operators, each of which took a difficulty argument and a number argument.  Or you could make one of the arguments the *type* of cap, which would let you use one operator to handle all the different types

Reply:
I arranged for it to apply only to the current difficulty. Honestly I'd prefer not to make changes that alter the argument configuration if I can get away with it, just because it's fiddly and I'm still not very familiar with the SEXP code. Separately, I'm a little skeptical that applying different damage caps for different difficulties on the fly in-mission is ever likely to be relevant.

Still, applying it only to the current difficulty feels more elegant to my intuitions than applying it to all, so here we are.

Reply:
Current difficulty works 👍